lib/chain.js

```diff
@@ -0,0 +1,107 @@
+'use strict'
+
+const doInject = require('./do-inject')
+
+const errorMessage = 'The dispatch function has already been invoked'
+
+class Chain {
```

Comment:
This refactor could be another PR - smaller PRs helps to get the code merged faster